### PR TITLE
flint-checker 0.0.4 (new formula)

### PR DESCRIPTION
**Pull request by mvitz**
Add a new formula to install flint (https://github.com/pengwynn/flint).

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This formula name conflicts with `homebrew/science/flint` but I'm not sure which other name to choose.

### CODE REVIEW

File: Formula/flint.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    (buildpath/"src/github.com/pengwynn").mkpath
+    ln_sf buildpath, buildpath/"src/github.com/pengwynn/flint"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "-o", bin/"flint"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/flint --version", 0)
```
Comment (bfontaine) [inline Formula/flint.rb]: Please write a test that does something more substantial than e.g. `--version` or `--help`.

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/flint.rb (hunk 1 above)
Comment (mvitz) [inline Formula/flint.rb]: I added tests for error, warning and success case.

Comment (bfontaine) [inline Formula/flint.rb]: Thanks!

**Review by bfontaine — APPROVED** [review]

**Comment by fxcoudert** [comment]
`  * Formula name conflicts with homebrew/science/flint`

**Comment by mvitz** [comment]
I mentioned this in my first comment ;-)

Is there any strategy to resolve such name conflicts in a generic way? I have no idea which name to choose.

**Comment by fxcoudert** [comment]
There is a strategy: if the name is taken, you need another one. I suggest `flint-checker`, for example.

**Comment by mvitz** [comment]
Argh. Good idea. Sometimes the solution is to near to see.

Have the commit messages and the PR title to be updated as well?

**Comment by fxcoudert** [comment]
PR title, formula file name and class name, commit message. And please squash all commits into a single one.

**Comment by mvitz** [comment]
Renamed everything to flint-checker.

**Comment by fxcoudert** [comment]
Thanks @mvitz for the pull request!

**Pull request closed without merging**